### PR TITLE
fix: view fragments

### LINKED ISSUE #75: Template `viewData` not inherited
If I understood correctly, my example should have inherited data. But I only see the first ones.
I think there is a problem when using `extend()` and `include()`.
Since the normal behavior is to transfer data internally to the final template
<details>
  <summary>Examples</summary>

  ```php
  class Home extends AbstractController
  {
      protected $data = [];
      public function index(): string
      {
          $this->data = ['foo' => 'FOO'];
          // 1.
          // return ''
          //  . view('/bug/view1', $this->data + ['bar' => 'BAR'])
          //  . view('/bug/view2', $this->data + ['baz' => 'BAZ'])
          //  . view('/bug/view3', $this->data + ['too' => 'TOO'])
          //  ;
          // 2.
          return ''
              . view_fragment('/bug/view1', ['fragment1'], $this->data + ['bar' => 'BAR'])
              . view_fragment('/bug/view2', ['fragment2'], $this->data + ['baz' => 'BAZ'])
              . view_fragment('/bug/view3', ['fragment3'], $this->data + ['too' => 'TOO'])
              . view_fragment('/bug/include', ['include'], $this->data + ['inc' => 'INC'])
              // . view('/bug/include', $this->data + ['inc' => 'INC'])
          ;
      }
  }
  ```
  ```php
  // layout
  <!DOCTYPE html>
  <html lang="en">
  <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <title>Document</title>
  </head>
  <body>
      <?= $this->renderSection('content') ?>
      <?= $this->include('/bug/include') ?>
  </body>
  </html>
  ```
  ```php
  // view1
  <?= $this->extend('/bug/layout') ?>
  <?php $this->section('content') ?>
  <div style="background-color: lightpink;width:200px;height: 130px;">
      <b>View 1</b><br>
      <b>foo: </b> <?= isset($foo) ? 'YES' : 'NO' ?><br>
      <b>bar: </b> <?= isset($bar) ? 'YES' : 'NO' ?><br>
      <b>baz: </b> <?= isset($baz) ? 'YES' : 'NO' ?><br>
      <b>too: </b> <?= isset($too) ? 'YES' : 'NO' ?><br>
      <b>inc: </b> <?= isset($inc) ? 'YES' : 'NO' ?><br>
  </div>
  <?php $this->endSection() ?>
  <?php $this->fragment('fragment1') ?>
  <div style="background-color: lightpink;width:200px;height: 130px;">
      <b>Fragment 1</b><br>
      <b>foo: </b> <?= isset($foo) ? 'YES' : 'NO' ?><br>
      <b>bar: </b> <?= isset($bar) ? 'YES' : 'NO' ?><br>
      <b>baz: </b> <?= isset($baz) ? 'YES' : 'NO' ?><br>
      <b>too: </b> <?= isset($too) ? 'YES' : 'NO' ?><br>
      <b>inc: </b> <?= isset($inc) ? 'YES' : 'NO' ?><br>
  </div>
  <?php $this->endFragment() ?>
  ```
  ```php
  // view2, view3 similar
  <div style="background-color: lightgreen;width:200px;height: 130px;">
      <b>View 2</b><br>
      <b>foo: </b> <?= isset($foo) ? 'YES' : 'NO' ?><br>
      <b>bar: </b> <?= isset($bar) ? 'YES' : 'NO' ?><br>
      <b>baz: </b> <?= isset($baz) ? 'YES' : 'NO' ?><br>
      <b>too: </b> <?= isset($too) ? 'YES' : 'NO' ?><br>
      <b>inc: </b> <?= isset($inc) ? 'YES' : 'NO' ?><br>
  </div>
  <?php $this->fragment('fragment2') ?>
  <div style="background-color: lightgreen;width:200px;height: 130px;">
      <b>Fragment 2</b><br>
      <b>foo: </b> <?= isset($foo) ? 'YES' : 'NO' ?><br>
      <b>bar: </b> <?= isset($bar) ? 'YES' : 'NO' ?><br>
      <b>baz: </b> <?= isset($baz) ? 'YES' : 'NO' ?><br>
      <b>too: </b> <?= isset($too) ? 'YES' : 'NO' ?><br>
      <b>inc: </b> <?= isset($inc) ? 'YES' : 'NO' ?><br>
  </div>
  <?php $this->endFragment() ?>
  ```
  ```php
  // include
  <?php $this->fragment('include') ?>
  <div style="background-color: lightpink;width:200px;height: 130px;">
      <b>Include</b><br>
      <b>foo: </b> <?= isset($foo) ? 'YES' : 'NO' ?><br>
      <b>bar: </b> <?= isset($bar) ? 'YES' : 'NO' ?><br>
      <b>baz: </b> <?= isset($baz) ? 'YES' : 'NO' ?><br>
      <b>too: </b> <?= isset($too) ? 'YES' : 'NO' ?><br>
      <b>inc: </b> <?= isset($inc) ? 'YES' : 'NO' ?><br>
  </div>
  <?php $this->endFragment() ?>
  ```
  Variant 2, add `['inc' => 'INC']` data. No last `baz, too, inc`
  ```
  View 1
  foo: YES
  bar: YES
  baz: NO
  too: NO
  inc: NO
  Include Frag
  foo: YES
  bar: YES
  baz: NO
  too: NO
  inc: NO
  Fragment 1
  foo: YES
  bar: YES
  baz: NO
  too: NO
  inc: NO
  Fragment 2
  foo: YES
  bar: YES
  baz: YES
  too: NO
  inc: NO
  Fragment 3
  foo: YES
  bar: YES
  baz: YES
  too: YES
  inc: NO
  Include Frag
  foo: YES
  bar: YES
  baz: NO
  too: NO
  inc: NO
  ```
  If uncomment last `view()`. `baz, too, inc` appears
  ```
  // the same as above, but last 
  Include Frag
  foo: YES
  bar: YES
  baz: NO
  too: NO
  inc: NO
  Include Frag
  foo: YES
  bar: YES
  baz: YES
  too: YES
  inc: YES
  ```
</details>

### PR DESCRIPTION
**Pull request by michalsn**
This PR fixes view fragments:
* handling multiple fragments with the same name
* fragments now respect sections

Closes #75

### CODE REVIEW

File: src/View/View.php [hunk 1]
```diff
@@ -27,6 +28,116 @@ class View extends BaseView
...
+
+        if ($this->showFragments) {
+            echo sprintf('@[[fragmentEnd="%s"]]', $name);
+        }
+    }
+
+    /**
+     * Whether we should display fragments tags or not.
+     */
+    protected function showFragments(bool $display = true): RendererInterface
```
Comment (neznaika0) [inline src/View/View.php:68]: Why not `self` or `static`?

File: tests/CommonTest.php [hunk 2]
```diff
@@ -87,4 +87,70 @@ public function testViewFragmentFromLayout(): void
...
+        $expected = <<<'EOD'
+            <b>Fragment 1 (1)</b><br>
+                    <b>Fragment 1 (3)</b><br>
+
+            EOD;
+
+        $this->assertSame($expected, $result);
+    }
+
+    public function testHugeView(): void
```
Comment (neznaika0) [inline tests/CommonTest.php:143]: By adding a test for `view(),` the result is not obvious - the layout inside the section. Is this a problem with the parent `view()` ?
```php
    public function testHugeViewFragment(): void
    {
        $result = view_fragment('huge/view', ['fragment_one']);

        $expected = <<<'EOD'
            Fragment one (from "huge/view")
            Fragment one (from "huge/include")
            Fragment one (from "huge/layout")

            EOD;

        $this->assertSame($expected, $result);
    }

    public function testHugeView(): void
    {
        $result = view('huge/view');

        $expected = <<<'EOD'


            View top

            Fragment one (from "huge/view")

            # include start in "huge/view"
            Layout top

            Section (from "huge/view")

            Include top

            Fragment one (from "huge/include")

            Include bottom

            Fragment one (from "huge/layout")

            Layout bottom
            # include end in "huge/view"

            View bottom

            EOD;

        $this->assertSame($expected, $result);
    }
```

Comment (neznaika0) [reply tests/CommonTest.php:143]: Yes, bug framework

File: src/View/View.php (hunk 1 above)
Comment (michalsn) [reply src/View/View.php:68]: I wanted to follow the current convention used in the original `View` class: https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/View/View.php

Comment (neznaika0) [reply src/View/View.php:68]: Good. This is a notification in the IDE

**Comment by neznaika0** [comment]
Everything is fine for multi-fragments. 
I'll check on the project later. I have a problem with flash messages working.

After the request, you need to return the template and toasts from the flash for it. There was no data before PR, we need to review it again.

Will the clarification of the work be indicated in the documentation?
> Fragments, like sections, are complemented, that is, by specifying several fragments of the same name, they will be displayed as one. But the input data for them may differ depending on the order of output.

**Review by neznaika0 — APPROVED** [review]

File: tests/CommonTest.php (hunk 2 above)
Comment (michalsn) [reply tests/CommonTest.php:143]: If you place the content outside the section this is what you will get. I don't see any issues here.

**Review by neznaika0 — APPROVED** [review]

**Pull request merged**